src/SprykerSdk/Zed/AopSdk/Business/OpenApi/Builder/OpenApiCodeBuilder.php

```diff
@@ -0,0 +1,732 @@
+    {
+        foreach ($contents as $response) {
+            if (!isset($response->schema)) {
+                continue;
+            }
+            if ($response->schema instanceof Schema) {
+                $responses[$this->getClassNameFromSchema($response->schema)] = $this->getReponsePropertiesFromSchema($response->schema, []);
+            }
+            if ($response->schema instanceof Reference) {
+                $responses[$this->getClassNameFromReference($response->schema)] = $this->getReponsePropertiesFromReference($response->schema, []);
+            }
+        }
+
+        return $responses;
+    }
+
+    /**
+     * @param \cebe\openapi\spec\Operation $operation
+     *
+     * @return array
+     */
+    protected function getResponsesFromOperation(Operation $operation): iterable
+    {
+        return $operation->responses ?? [];
+    }
+
+    /**
+     * @param \cebe\openapi\spec\Schema $schema
+     * @param array $rootType
+     *
+     * @return array
+     */
+    protected function getReponsePropertiesFromSchema(Schema $schema, array $rootType): array
+    {
+        foreach ($this->getPropertiesFromSchema($schema) as $schemaObject) {
+            if ($schemaObject instanceof Schema && isset($schemaObject->properties) && !empty($schemaObject->properties)) {
+                $rootType[] = false;
+
+                return $this->getReponsePropertiesFromSchema($schemaObject, $rootType);
+            }
+            if ($schemaObject instanceof Reference && isset($schemaObject->properties) && !empty($schemaObject->properties)) {
+                $rootType[] = false;
+
+                return $this->getReponsePropertiesFromReference($schemaObject, $rootType);
+            }
+
+            if (!isset($schemaObject->items)) {
+                continue;
+            }
+
+            if ($schemaObject->items instanceof Schema && isset($schemaObject->items->properties) && !empty($schemaObject->items->properties)) {
+                $rootType[] = true;
+
+                return $this->getReponsePropertiesFromSchema($schemaObject->items, $rootType);
+            }
+
+            if ($schemaObject->items instanceof Reference && isset($schemaObject->items->properties) && !empty($schemaObject->items->properties)) {
+                $rootType[] = true;
+
+                return $this->getReponsePropertiesFromReference($schemaObject->items, $rootType);
+            }
+        }
+        if (current($rootType) === true) {
+            return $this->formatPropertiesToArrayOfClassInstance($this->getClassNameFromSchema($schema));
+        }
+
+        return $this->formatResponseProperties($this->getPropertiesFromSchema($schema));
+    }
+
+    /**
+     * @param \cebe\openapi\spec\Reference $schema
+     * @param array $rootType
+     *
+     * @return array
+     */
+    protected function getReponsePropertiesFromReference(Reference $schema, array $rootType): array
```

Comment:
This method and the method above look exactly the same. Can you reduce it to one method? You could omit the argument type hint and set it as `@param \cebe\openapi\spec\Schema|\cebe\openapi\spec\Reference $schemaOrReference` in the method doc block.